File: exercise/php/day07/src/Workshop/Workshop.php
<?php

namespace Workshop;
class Workshop
{
    private $gifts = [];

    public function addGift(Gift $gift): void
    {
        $this->gifts[] = $gift;
    }

    public function completeGift(string $name): ?Gift
    {
        foreach ($this->gifts as $gift) {
            if ($gift->getName() === $name) {
                return $gift->withStatus(Status::PRODUCED);
            }
        }

        return null;
    }
}

File: exercise/C#/day07/Workshop/Workshop.cs
namespace Workshop
{
    public enum Status
    {
        Producing,
        Produced
    }

    public record Gift(string Name, Status Status = Status.Producing);

    public class Workshop
    {
        private readonly List<Gift> _gifts = [];

        public void AddGift(Gift gift) => _gifts.Add(gift);

        public Gift? CompleteGift(string name)
        {
            var gift = _gifts.FirstOrDefault(g => g.Name == name);
            return gift != null
                ? gift with {Status = Status.Produced}
                : null;
        }
    }
}